src/pages/projectSettings/tabs/PluginsTab/PluginsTab.tsx

```diff
@@ -72,5 +73,21 @@ const validate: ValidateProps<FormState> = (formData, errors) => {
     );
   }
 
+  const displayNameDefined = !!patchMetadataPanelLink.displayName.trim();
+  const urlTemplateDefined = !!patchMetadataPanelLink.urlTemplate.trim();
+  if (displayNameDefined && !urlTemplateDefined) {
+    errors.externalLinks.patchMetadataPanelLink.urlTemplate.addError(
+      "You must specify a URL template or exclude display name."
+    );
+  } else if (!displayNameDefined && urlTemplateDefined) {
+    errors.externalLinks.patchMetadataPanelLink.displayName.addError(
+      "You must specify a display name or exclude URL template."
+    );
+  }
+  if (patchMetadataPanelLink.displayName.length > 40) {
+    errors.externalLinks.patchMetadataPanelLink.displayName.addError(
+      "Display name must be less than 40 characters."
+    );
+  }
```

Comment:
RJSF can handle this automatically if you pass in the maxLength property to this field in the schema definition.